### ISSUE TITLE
`unused-async` (`RUF029`) - false negative on methods

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#24785]
## [Playground](https://play.ruff.rs/df26db3e-dc57-4ca9-842e-7e96e0a4ac1c)

```py
from typing import final

def foo(): ...

async def bar(): # error
    foo()

class Foo:
    async def foo(self): # no error
        bar()

```

### Version

0.15.11

**Comment by MichaReiser** [comment]
This seems to have been somewhat intentional, see https://github.com/astral-sh/ruff/pull/9966#issuecomment-1939953539

But I think we could change it in preview to flag methods for as long as the class doesn't have any base class or isn't attributed with `@overriden`

**Comment by ntBre** [comment]
I found https://github.com/astral-sh/ruff/issues/10980, which I believe is tracking the same issue and also links to the previous discussion on #9966 that Micha mentioned.
